src/rdstring.c

```diff
@@ -553,6 +553,135 @@ char **rd_string_split(const char *input,
         return arr;
 }
 
+/**
+ * @brief Split a character-separated string into an array, supporting quoted
+ * fields.
+ *
+ * @remark This function is designed to handle cases where fields may be
+ * enclosed in double quotes (`"`), allowing separators within quoted fields to
+ * be treated as part of the field rather than as delimiters.
+ *
+ * @param input Input string to parse.
+ * @param sep The separator character (typically ',').
+ * @param skip_empty Do not include empty fields in the output array.
+ * @param cntp Will be set to the number of elements in the array.
+ *
+ * Supports quoted fields using `"`. Escaped characters (e.g., `\n`, `\t`) are
+ * also handled. Quoted fields can contain separators, and the quotes will be
+ * removed in the output. The array and the array elements will be allocated
+ * together and must be freed with a single rd_free(array) call. The array
+ * elements are copied, and any escape sequences or quotes are removed.
+ *
+ * @returns The parsed fields in an array. The number of elements in the
+ *          array is returned in \p cntp.
+ */
+char **rd_string_split_csv(const char *input,
```

Comment:
I think we should return error code and also handle the cases where input string is invalid, like `"` started but no ending `"` present.